### DISCUSSION TITLE
First Timer Contribution Idea

### DISCUSSION THREAD

**Discussion by madisonbrading** [PrairieLearn/PrairieLearn#8662] [Q&A]
I have never contributed to an open source project before and am wondering if anyone could help guide me through how to find an issue to work on so I can contribute! Thank you

**Comment by nwalters512** [comment]
Welcome! In the past, we've tagged some of our issues with the ["good first issue" label](https://github.com/PrairieLearn/PrairieLearn/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). That might be a decent place to start, though we admittedly haven't done a great job of keeping those labels up to date. You're also more than welcome to look at all the issues and see if any look interesting to you.

**Comment by madisonbrading** [comment]
ok thank you. some seem a tad old / looks like there is nothing for someone to do ie: https://github.com/PrairieLearn/PrairieLearn/issues/3162

but one I find interesting is this: https://github.com/PrairieLearn/PrairieLearn/issues/3162 and I think I could start with pl-matrix-input or pl-number-input! Let me know your thoughts... also wondering if you recommend I use github desktop in combination with VScode to add my changes

**Reply by nwalters512** [reply]
Perhaps you meant to link to a different issue in your second paragraph? For what it's worth, #3162 is still relevant despite being old - many of our issues have been around for a long time but still need to be addressed!

GitHub Desktop is fine, and most PL devs use VSCode, so we can definitely recommend it!

**Reply by madisonbrading** [reply]
yes meant https://github.com/PrairieLearn/PrairieLearn/issues/6968

**Reply by nwalters512** [reply]
Yep, that sounds like a good one! I just checked and `pl-number-input` since gained a `placeholder` attribute, which leaves `pl-matrix-input`. You can drop a comment on the issue saying you'll work on that.

**Reply by eliotwrobson** [reply]
@madisonbrading That issue hadn't been updated, but pl-number-input had the custom placeholder logic added a little while back. I think the only element in #6968 that still needs a custom placeholder is pl-matrix-input. Though that element code doesn't have any tooling set up at the moment, so it may be more challenging to work on than other elements (though you're welcome to help set that up if you wish!).

EDIT: tl;dr what Nate said is right, we were typing at the same time.

**Reply by madisonbrading** [reply]
Thanks! and for https://github.com/PrairieLearn/PrairieLearn/issues/3162 I have never done constraints but was planning to add the following code in PrairieLearn/apps/prairielearn/src/migrations/20200818200925_resilient_syncing.sql

ALTER TABLE questions
DROP CONSTRAINT questions_course_id_uuid_key;

not sure if I need to add the "WHERE deleted_at IS NULL" part of the constraint when I try to delete! Also - once I make this change do I just attempt to push and request review?

**Reply by jonatanschroeder** [reply]
Hi, @madisonbrading. When changing the database, you should create a new migration, not change an existing one. The existing ones are expected to run only once, so modifying them will not cause them to run again.

And when dropping constraints you don't need to specify its limitations, dropping by name should be enough.

I encourage you to try and create a new PR with your changes, it's easier to provide feedback there. Note that you can make further changes after your first submission.
